gmt/utils.py

```diff
@@ -180,3 +183,64 @@ def is_nonstr_iter(value):
     except TypeError:
         is_iterable = False
     return bool(not isinstance(value, str) and is_iterable)
+
+
+class GMTTempFile():
+    """
+    Context manager for temporaray file.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> with GMTTempFile() as tmpfile:
+    ...     # write data to temporary file
+    ...     x = y = z = np.arange(0, 3, 1)
+    ...     np.savetxt(tmpfile.name, (x, y, z), fmt='%.1f')
+    ...     lines = tmpfile.read()
+    ...     print(lines)
+    ...     nx, ny, nz = tmpfile.loadtxt(unpack=True, dtype=float)
+    ...     print(nx, ny, nz)
+    0.0 1.0 2.0
+    0.0 1.0 2.0
+    0.0 1.0 2.0
+    <BLANKLINE>
+    [0. 0. 0.] [1. 1. 1.] [2. 2. 2.]
+    """
+    def __init__(self, prefix="gmt-python-", suffix=".txt"):
+        with NamedTemporaryFile(prefix=prefix, suffix=suffix) as tmpfile:
+            self.name = tmpfile.name
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        if os.path.exists(self.name):
+            os.remove(self.name)
+
+    def read(self, keep_tabs=False):
+        """
+        Read the entire contents of the file.
```

Comment:
Please add "as a Unicode string" to the description. This will help distinguish it from `loadtxt`.